libzecale/tests/circuits/fp12_2over3over2_gadgets_test.cpp

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2015-2020 Clearmatics Technologies Ltd
+//
+// SPDX-License-Identifier: LGPL-3.0+
+
+#include "libzecale/circuits/fields/fp12_2over3over2_gadgets.hpp"
+#include "libzecale/circuits/pairing/bls12_377_pairing.hpp"
+#include "libzecale/circuits/pairing/bw6_761_pairing_params.hpp"
+
+#include <gtest/gtest.h>
+#include <libff/algebra/curves/bls12_377/bls12_377_pp.hpp>
+#include <libff/algebra/curves/bw6_761/bw6_761_pp.hpp>
+#include <libzeth/snarks/groth16/groth16_snark.hpp>
+
+using ppp = libff::bw6_761_pp;
+using snark = libzeth::groth16_snark<ppp>;
+
+namespace
+{
+
+TEST(Fp12_2over3over2_Test, SquareGadgetTest)
+{
+    using Fp12T = libff::bls12_377_Fq12;
+    using FieldT = typename Fp12T::my_Fp;
+    using Fp2T = typename Fp12T::my_Fp2;
+    using Fp6T = typename Fp12T::my_Fp6;
+
+    // Native squaring
+    const Fp12T z(
+        Fp6T(
+            Fp2T(FieldT("5"), FieldT("6")),
+            Fp2T(FieldT("7"), FieldT("8")),
+            Fp2T(FieldT("9"), FieldT("10"))),
+        Fp6T(
+            Fp2T(FieldT("21"), FieldT("22")),
+            Fp2T(FieldT("23"), FieldT("24")),
+            Fp2T(FieldT("25"), FieldT("26"))));
+    const Fp12T z_squared = z.squared();
+
+    // Squaring in a circuit
+    libsnark::protoboard<FieldT> pb;
+    libzecale::Fp12_2over3over2_variable<Fp12T> z_var(pb, "z");
+    libzecale::Fp12_2over3over2_variable<Fp12T> z_squared_var(pb, "z");
+    const size_t num_primary_inputs = pb.num_inputs();
+    pb.set_input_sizes(num_primary_inputs);
+
+    libzecale::Fp12_2over3over2_square_gadget<Fp12T> square_gadget(
+        pb, z_var, z_squared_var, "square z");
```

Comment:
We use `<space>` as an annotation separator. As such, we should not use spaces in annotations (I read your annotation as "the variable z of the gadget square"). let's use an underscore to have a snake_case name and resolve this.

Reply:
added a space